File: 프로그래머스/Lv.1/12977. 소수 만들기/소수 만들기.py
def gen_prime_table(max_num) : 
    prime_table = [
        2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 
        43, 47, 53, 59, 61, 67, 71, 73, 79,83, 89, 97
    ]

    for number in range(100, max_num + 1) : 
        flag = True
        for check_num in prime_table : 
            if number % check_num == 0 : 
                flag = False
                break
        
        if flag : 
            prime_table.append(number)    
    
    return prime_table

def solution(nums):
    answer = 0
    prime_table = gen_prime_table(1_000 * 3)

    for i in range(len(nums) - 2) : 
        for j in range(i + 1, len(nums) - 1) : 
            for k in range(j + 1, len(nums)) : 
                number = nums[i] + nums[j] + nums[k]
                if number in prime_table : 
                    answer += 1

    return answer

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java

class Solution {
    public int solution(int[] nums) {
        // false : prime        |   true : not prime
        boolean[] primeTable = new boolean[3_000 + 1];
        primeTable[0] = primeTable[1] = true;
        int sqrt = (int) Math.sqrt(3_000);

        for (int i = 2; i <= sqrt; i++)     if (!primeTable[i])
        for (int j = 2 * i; j <= 3_000; j += i)
        primeTable[j] = true;
        
        int answer = 0;
        int len = nums.length;

        for (int i = 0; i < len - 2; i++)
        for (int j = i + 1; j < len - 1; j++)
        for (int k = j + 1; k < len; k++)
        if (!primeTable[nums[i] + nums[j] + nums[k]])
        answer++;

        return answer;
    }
}